cypress-io/cypress
# feat: add cloud recommendation message to CI output

- Closes https://github.com/cypress-io/cypress/issues/24000

### User facing changelog


n/a

### Additional details


This PR adds a message that prompts the user to go to Cypress Cloud after the test summary table only when **all** of the following conditions are true:
- Cypress is in run mode
- The process is running in CI (using `is-ci` library to determine this)
- `CYPRESS_COMMERCIAL_RECOMMENDATIONS` environment variable is **not** set to `0`
- At least one test in the run failed
- `--record` flag was not passed to the cypress run command (the user is not attempting to record this run to Cypress Cloud)

Services PR for on link: https://github.com/cypress-io/cypress-services/pull/4972
Docs PR: https://github.com/cypress-io/cypress-documentation/pull/4863

### Steps to test


Look at the output from [this run](https://app.circleci.com/pipelines/github/cypress-io/cypress/45747/workflows/fc2c3edc-752b-4adc-887a-0fc9cc3d31e4/jobs/1924189) and verify that the cloud recommendation message appears after the test summary. Note that Circle CI is redacting `cypress.io` from the on link. This is a setting in our Circle CI project and shouldn't be an issue in other's projects.

You can also look at the tests that I added in `cypress_spec.js` and run those

### How has the user experience changed?


![Screen Shot 2022-11-17 at 2 46 33 PM](https://user-images.githubusercontent.com/14275198/202544303-005030f9-331b-44f9-9439-364f96a957a9.png)

### PR Tasks


- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/4863
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
